subsys/bluetooth/mesh/gen_ponoff_internal.h

```diff
@@ -0,0 +1,25 @@
+/*
+ * Copyright (c) 2021 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
+ */
+
+/**
+ * @file
+ * @defgroup bt_mesh_ponoff_internal Generic Power OnOff model internals
+ * @ingroup bt_mesh_ponoff
+ * @{
+ * @brief Common internal API for the Generic Power OnOff models.
+ */
+
+#ifndef BT_MESH_GEN_PONOFF_INTERNAL_H__
+#define BT_MESH_GEN_PONOFF_INTERNAL_H__
+
+enum {
```

Comment:
As a high-level solution, it is Ok.
However, I think, it is better to have access layer functionality to check whether the model is part of the extensions. Since the extension implementation is the left-child right-sibling binary tree it is possible to understand whether the model is the part of a tree and which range. 
In this case, we do not need some additional substates for models both here and DTT changes that Pavel made.
This checking will be classic work with the tree inside access rather than the unique solution for every model.

Reply:
@trond-snekvik , what do you think?

Reply:
It could work, but it needs to be documented properly, as it's an unexpected side effect for these models. We'd also need to force extensions to be enabled for our models.

Reply:
IMHO make model extension as the optional feature is the arguable solution since this is the fundamental mandatory feature of the bluetooth model spec.
We still have this side effect but implemented it as an internal API.

Reply:
You can't use `bt_mesh_model` struct in this case. When `init` callback is called, the extended model may not have model pointer set yet. That is why e.g. `bt_mesh_model_extend` call is moved to `start` callback for Light LC Server model: https://github.com/nrfconnect/sdk-nrf/blob/master/subsys/bluetooth/mesh/light_ctrl_srv.c#L1508-L1515
This may also cause problems if you implement a vendor model which extends Generic Power OnOff. It isn't be clear how to extend it in this case.

Reply:
As I understand using API before initialization is not an implementation problem. That means to the moment of power-up functionality `bt_mesh_model` structure is filled and ready, isn't?

Reply:
Light LC server extends Light Lightness server at its `start` callback. Which means that when Light Lightness server `start` callback is called it may not know yet whether it is extended or not.

Reply:
Yep, I see. I still think it is the general issue not only PowerOnOff or LL extension of LC. There is bug with Scene usage of extended models (NCSDK-8536).
Need to discuss in detail how to solve such issues generally. It is better to do outside this PR.

Reply:
These are 2 different issues. While power up sequence happens at the start of the mesh stack, scene recall is happening after the mesh stack is started, which makes the extend mechanism suitable for scene.